### ISSUE TITLE
GCop607 false alarm

### ISSUE THREAD

**Issue by sungam3r** [Geeksltd/GCop#258]
Usage:
```
  <ItemGroup>
    <PackageReference Include="GCop.All.Common" Version="2.6.0" PrivateAssets="all" />
  </ItemGroup>
```

Example:

```csharp
  internal static class Extensions
    {
        public static IEnumerable<MethodInfo> GetNormalMethods(IEnumerable<MethodInfo> methods)
        {
            return methods.Where(m => !m.IsSpecialName); // GCop607	It should be written as methods.Except(m => m.IsSpecialName)	
        }
    }
```

But obviously it won't compile:
```csharp
 internal static class Extensions
    {
        public static IEnumerable<MethodInfo> GetNormalMethods(IEnumerable<MethodInfo> methods)
        {
            return methods.Except(m => m.IsSpecialName); // CS1660	Cannot convert lambda expression to type 'IEnumerable<MethodInfo>' because it is not a delegate type	
        }
    }
```

**Comment by sungam3r** [comment]
Still have this error in 2.6.1

**Comment by sungam3r** [comment]
@Karvan Any progress on this?
